api/bugzilla/api_bugzilla.py

```diff
@@ -81,8 +244,181 @@ def __init__(self):
+        if not bug_ids:
+            return pd.DataFrame()
+
+        include_fields = ["id", "type", "cf_qa_whiteboard", "resolution",
+                          "keywords", "severity"] + version_fields
+        rows = []
+
+        for i in range(0, len(bug_ids), batch_size):
+            batch = bug_ids[i:i + batch_size]
+            query = {"id": ",".join(map(str, batch)), "include_fields": include_fields}
+            bugs = BugzillaHelper().query(query)
+
+            for bug in bugs:
+                for fname in version_fields:
+                    m = FIREFOX_FLAG_STATUS_VERSION.match(fname)
+                    if not m:
+                        continue
+                    version = int(m.group(1))
+                    raw = getattr(bug, fname, None)
+                    status = (str(raw).strip().lower() if raw and str(raw).strip() else '---') # noqa
+
+                    rows.append({
+                        "bugzilla_key": int(bug.id),
+                        "type": bug.type,
+                        "flag-version": version,
+                        "status": status,
+                        "keywords": ", ".join(bug.keywords),
+                        "severity": bug.severity,
+                        "qa-found-in": getattr(bug, "cf_qa_whiteboard", ""),
+                        "resolution": getattr(bug, "resolution", None)
+                    })
+
+        df = pd.DataFrame(rows)
+        if df.empty:
+            return df
+
+        # Only fetch histories for fixed/verified flags
+        need_mask = df["status"].isin(["fixed", "verified"])
+        candidate_ids = (
+            df.loc[need_mask, "bugzilla_key"]
+              .dropna().astype(int).unique().tolist()
+        )
+
+        if candidate_ids:
+            # Concurrent history fetch (keeps results in memory)
+            with requests.Session() as sess:
+                hist_cache = self.BugzillaHelperClient.fetch_many_bug_histories(
+                    candidate_ids,
+                    session=sess,
+                    max_workers=hist_workers,
+                    batch_size=hist_batch_size,
+                    timeout=hist_timeout,
+                    retries=hist_retries,
+                    stream_callback=None,  # return dict
+                )
+
+            # Precompute version->first_ts per bug
+            first_ts_map = {
+                bid: self.first_fixed_verified_by_version(hist_cache.get(bid, []))
+                for bid in candidate_ids
+            }
+
+            def compute_row_ts(row):
+                if row["status"] not in ("fixed", "verified"):
+                    return pd.NaT
+                bid = int(row["bugzilla_key"])
+                v = int(row["flag-version"])
+                return first_ts_map.get(bid, {}).get(v, pd.NaT)
+
+            df["bugzilla_flag_fixed_at"] = df.apply(compute_row_ts, axis=1)
+        else:
+            df["bugzilla_flag_fixed_at"] = pd.NaT
+
+        if save_csv:
```

Comment:
Will be removed, leaving it for debugging purposes if need be while finishing this work